### ISSUE TITLE
RFE: Visualization of expected EQ frequency response

### ISSUE THREAD

**Issue by vmatare** [wwmm/pulseeffects#663]
Hey, I have an idea for a nice feature:

The EQ could really benefit from a visualization of the expected frequency response. Hz on the X axis and DB offset on Y.

So the background is I have a bi-amping setup which gives plenty of headroom in the bass range, but the limiting factor on getting an awesome bass out of it are the [normal modes](https://en.wikipedia.org/wiki/Normal_mode) of my living room.

There is a bunch of those, and I can identify their peak frequency okayish, but I'd like to be more precise when trying to cout them out. Currently, I search through the frequencies with a sine wave and listen for where it gets out of hand. Try to smooth it out and then listen to the surrounding frequency range, find gaps, tune again and so forth. Slightly tedious.

So I think it might be pretty helpful to at least see what I'm changing, because listening takes much more time, and people also tend to get seasick from those sine waves.

I haven't tried "tuning" the room with a mic yet, but it's definitely something I'd like to try later on. I have a feeling pulseeffects can help me with that as well...

**Comment by wwmm** [comment]
I would like this too. But as you can see here #515 it is hard to get that when you do not use the native plugin graphical interface

**Comment by vmatare** [comment]
Oh yeah, I imagined the function described by the equalizer might be hard to find/model. So I thought, why not just measure it? Sample a sine through the spectrum and record the difference between input and output volume, bam. Should be possible to do this within a few seconds, maybe even less than one if the FFT and whatever magic is going on there converges quickly enough...

**Comment by vmatare** [comment]
Maybe it works just as well (or even better) with noise. That should be much faster.

**Comment by vmatare** [comment]
Oh, this is an interesting read: https://lsp-plug.in/?page=manuals&section=para_equalizer_x32_lr
So we could even measure the EQ's quality in terms of impulse response... o_O

**Comment by lee-tts** [comment]
@vmatare Did you manage to tame your room modes finally? I am here for this exact reason myself, and have one that's especially bad at 116Hz that I wanted to squash with pulseeffects. I realize that in the current version it's no longer possible to change the bandwidth / width of the EQ like in the past. Pity because my room has pretty sharp spikes and the EQ is a bit wide on a given parameter.

**Comment by wwmm** [comment]
@lee-tts you can change the bandwidth. You even have more filters available like notch filters

**Comment by vmatare** [comment]
Yeah, you now have this "Quality" control which affects the bandwidth. But to answer you question @lee-tts, I wouldn't say I've tamed anything finally. I have this particularly bad resonance around 85 Hz. The EQ band at that spot is set to -21.6 dB, but that's for modern mixes. Seems to me like old mixes (like music from 60s to 70s) anticipated more bass resonance. Anyways, there are other resonances as well, and while I feel I have cut out the worst of it, but have created some holes in the process. I.e. I'm cutting out too wide, and in other low frequency ranges I still notice minor resonances and gaps now and then.

So yeah, definitely far from perfect, but at least I'm getting some of that impressive ultra-low bass you can have nowadays.

**Comment by lee-tts** [comment]
Thanks guys.

So the quality somehow affects the bandwidth? I played with it a bit initially but the 'width' written in the middle of the dialog didn't change so I moved on. 

I've got one parameter set to 116Hz and lowering it more than about 6dB basically scoops all the bass in the music I'm listening to (classic rock, metal from all eras -- there's some classic rock that has plenty of 116Hz haha that Bb2 is deadly)

**Comment by wwmm** [comment]
You have a good read here https://en.wikipedia.org/wiki/Q_factor. Long story short `bandwidth = center_frequency / quality_factor`

**Comment by lee-tts** [comment]
@wwmm Got it! Thank you. Not sure why I didn't see the width number changing before. Now I see that it does change when you adjust quality.

**Comment by wwmm** [comment]
We can close this issue. Nowadays we are able to show LSP native window.
